### DISCUSSION TITLE
Dagster  used for long-running tasks like consuming Kafka data

### DISCUSSION THREAD

**Discussion by woestler** [dagster-io/dagster#21428] [Q&A] [answered]
Can dagster be used for long-running extraction such as realtime Kafka consumption?
I searched for this question on Google, but I didn't find any definitive answers, and I'm very interested in this matter, hoping someone can come and answer this question.

**Comment by garethbrickman** [comment, marked as answer]
Dagster doesn't currently have an integration for consuming Kafka data but it has been requested in [https://github.com/dagster-io/dagster/issues/11857](https://github.com/dagster-io/dagster/issues/11857)

You could consider using [Dagster Pipes](https://docs.dagster.io/guides/dagster-pipes) as an approximate solution, depending on your use case.

**Comment by nitinsinghh27** [comment]
Hey @woestler 
Just wanted to check, did you find any work around?

**Reply by cmpadden** [reply]
Hey @nitinsinghh27 - you can find a demo project here that shows how to use sensors to poll a Kafka topic and produce Dagster run requests.

https://github.com/cmpadden/dagster-kafka-demo/blob/main/dagster_kafka_demo/sensors.py
